server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -648,8 +657,12 @@ public void updateState(ClusterState state) {
         metadata = getRepoMetadata(state);
         final Settings updatedSettings = metadata.settings();
         if (updatedSettings.equals(previousSettings) == false) {
-            snapshotRateLimiter = getRateLimiter(metadata.settings(), MAX_SNAPSHOT_BYTES_PER_SEC);
-            restoreRateLimiter = getRateLimiter(metadata.settings(), MAX_RESTORE_BYTES_PER_SEC);
+            snapshotRateLimiter = getRateLimiter(
```

Comment:
I think we also want to update the snapshot rate limiter when the recovery settings is updated too? Here it is only updated when `(updatedSettings.equals(previousSettings) == false)`